src/jdb_to_nwb/convert_photometry.py

```diff
@@ -1,22 +1,199 @@
+    W=diags(w,0,shape=(m,m))
+    A=csc_matrix(W+(lambda_*D.T*D))
+    B=csc_matrix(W*X.T)
+    background=spsolve(A,B)
+    return np.array(background)
+
+def airPLS(x, lambda_=100, porder=1, itermax=15):
+    '''
+    Adaptive iteratively reweighted penalized least squares for baseline fitting
+    
+    input
+        x: input data (i.e. chromatogram of spectrum)
+        lambda_: parameter that can be adjusted by user. The larger lambda is,  the smoother the resulting background, z
+        porder: adaptive iteratively reweighted penalized least squares for baseline fitting
+    
+    output
+        the fitted background vector
+    '''
+    m=x.shape[0]
+    w=np.ones(m)
+    for i in range(1,itermax+1):
+        z=WhittakerSmooth(x,w,lambda_, porder)
+        d=x-z
+        dssn=np.abs(d[d<0].sum())
+        if(dssn<0.001*(abs(x)).sum() or i==itermax):
+            if(i==itermax): print('WARING max iteration reached!')
+            break
+        w[d>=0]=0 # d>0 means that this point is part of a peak, so its weight is set to 0 in order to ignore it
+        w[d<0]=np.exp(i*np.abs(d[d<0])/dssn)
+        w[0]=np.exp(i*(d[d<0]).max()/dssn) 
+        w[-1]=w[0]
+    return z
 
 
 def add_photometry(nwbfile: NWBFile, metadata: dict):
     print("Adding photometry...")
     # get metadata for photometry from metadata file
     signals_mat_file_path = metadata["photometry"]["signals_mat_file_path"]
-    photometry_sampling_rate_in_hz = metadata["photometry"]["sampling_rate"]
+    # photometry_sampling_rate_in_hz = metadata["photometry"]["sampling_rate"] Commented out to use the old sampling rate variable
 
     # TODO: extract photometry signals
+    # Jose:
+    # I copied over the code most relevant to the TODO from Python scripts, assuming MATLAB is already convered the binary data. 
+
+    signals = scipy.io.loadmat(signals_mat_file_path,matlab_compatible=True)
+    # used to be : signals = scipy.io.loadmat(datepath+'signals.mat',matlab_compatible=True) 
+
+    # raw signals are saved as ndarrays 
+    ref = signals['ref']
+    sig1 = signals['sig1']
+    sig2 = signals['sig2']
+
+    SR = 10000 #sampling rate of photometry system
+    Fs = 250 #downsample frequency 
+
+    #downsample your raw data from 10kHz to 250Hz. 
+    #Takes every 40th sample and stores it as sig1,sig2, or ref
+    sig1 = sig1[0][::int(SR/Fs)]
+    ref = ref[0][::int(SR/Fs)]
+    sig2 = sig2[0][::int(SR/Fs)]
+
+    sigs = pd.DataFrame()
+    sigs['ref'] = ref
+    sigs['green'] = sig1
+    sigs['red'] = sig2
+
+    # define reference and signal traces for analysis
+    raw_reference = sigs['ref']
+    raw_green = sigs['green']
+    raw_red = sigs['red']
+
+    smooth_window = int(Fs/30)
+    # smooth signals and plot
+    reference = np.array(raw_reference.rolling(window=smooth_window,min_periods=1).mean()).reshape(len(raw_reference),1) 
+    # this finds the rolling average of the signal given the time window of 250Hz/30(30 second bins)
```

Comment:
This creates a smoothing window of 8 samples, which is not 30 second bins. 8 samples = 0.032 second bins. Either `Fs/30` should be `Fs*30` or the comment should be changed.